src/osp/Active/drawing.h

```diff
@@ -99,25 +88,59 @@ struct MaterialData
     std::vector<ActiveEnt>  m_removed;
 };
 
+using MeshRefCount_t    = lgrn::IdRefCount<MeshId>;
+using MeshIdOwner_t     = MeshRefCount_t::Storage_t;
+
+using TexRefCount_t     = lgrn::IdRefCount<TexId>;
+using TexIdOwner_t      = TexRefCount_t::Storage_t;
+
 /**
- * @brief Storage for Drawing components
+ * @brief Mesh Ids, texture Ids, and storage for drawing-related components
  */
 struct ACtxDrawing
 {
+
+    // Drawing Components
     acomp_storage_t<ACompOpaque>            m_opaque;
     acomp_storage_t<ACompTransparent>       m_transparent;
     acomp_storage_t<ACompVisible>           m_visible;
     acomp_storage_t<ACompColor>             m_color;
 
-    acomp_storage_t<ACompMesh>              m_mesh;
-    std::vector<osp::active::ActiveEnt>     m_meshDirty;
+    // Data needed by each unique material Id: assigned entities and
+    // added/removed queues. Access with m_materials[material ID]
+    std::vector<MaterialData>               m_materials;
 
-    acomp_storage_t<ACompTexture>           m_diffuseTex;
+    // Scene-space Meshes
+    lgrn::IdRegistry<MeshId>                m_meshIds;
+    MeshRefCount_t                          m_meshRefCounts;
+
+    // Scene-space Textures
+    lgrn::IdRegistry<TexId>                 m_texIds;
+    TexRefCount_t                           m_texRefCounts;
+
+    // Meshes and textures assigned to ActiveEnts
+    acomp_storage_t<TexIdOwner_t>           m_diffuseTex;
     std::vector<osp::active::ActiveEnt>     m_diffuseDirty;
 
-    // Access with m_materials[material ID]
-    std::vector<MaterialData>               m_materials;
+    acomp_storage_t<MeshIdOwner_t>          m_mesh;
+    std::vector<osp::active::ActiveEnt>     m_meshDirty;
 };
 
+/**
+ * @brief Associates mesh/texture resources Ids from ACtxDrawing with Resources
+ */
+struct ACtxDrawingRes
+{
+    // Associate Texture Ids with resources
+    IdMap_t<ResId, TexId>                   m_resToTex;
+    IdMap_t<TexId, ResIdOwner_t>            m_texToRes;
+
+    // Associate Mesh Ids with resources
+    IdMap_t<ResId, MeshId>                  m_resToMesh;
+    IdMap_t<MeshId, ResIdOwner_t>           m_meshToRes;
+};
+
+//
+
```

Comment:
Not necessary, but I'm considering moving these to a new header. This will allow ACtxDrawing to be included without needing include resourcetypes.h.